File: c#/src/Docmost.Sdk/Models/UpdateSpaceMemberRoleDto.cs
namespace Docmost.Sdk.Models;

public class UpdateSpaceMemberRoleDto
{
    [System.Text.Json.Serialization.JsonPropertyName("spaceId")]
    public string SpaceId { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("userId")]
    public string? UserId { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("groupId")]
    public string? GroupId { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("role")]
    public string Role { get; set; }

}

File: go/docmost/models/update_space_member_role_dto.go
package models

// UpdateSpaceMemberRoleDto from OpenAPI.
type UpdateSpaceMemberRoleDto struct {
	SpaceId string `json:"spaceId"`
	UserId *string `json:"userId,omitempty"`
	GroupId *string `json:"groupId,omitempty"`
	Role string `json:"role"`
}
